iceoryx_posh/source/roudi/shared_memory_manager.cpp

```diff
@@ -435,6 +417,49 @@ void SharedMemoryManager::deleteRunnableAndItsPorts(std::string f_runnableName)
     }
 }
 
+void SharedMemoryManager::destroySenderPort(SenderPortType::MemberType_t* const senderPortData)
```

Comment:
don't pass a MemberType but a SenderPortType, this makes it possible to handle PortExtensions which inherit from SenderPortType

Reply:
But I have to use the embedded MemberType to remove it from the list. The getter for this is currently protected. The MemberType is already an abstraction that every PortType provides, or?